src/include/migraphx/op/pooling.hpp

```diff
@@ -112,7 +136,7 @@ struct pooling
         const shape& input = inputs.at(0);
         auto padding_size  = padding.size();
         size_t kdims       = input.ndim() - 2;
-        if(input.ndim() != padding_size / 2 + 2 and input.ndim() != padding_size + 2)
+        if(input.ndim() * 2 != padding_size + 4 and input.ndim() != padding_size + 2)
```

Comment:
Are we not concerned of overflowing ndim() since we're now shifting it up a bit in size? 

Also, i suppose if ndim == 2  and padding_size == 0, this would trigger? Is that the intent here?

Reply:
--Are we not concerned of overflowing ndim() since we're now shifting it up a bit in size?

No, it represents the dimensions of a tensor and is a 64-bit size_t; I'd call it very safe to assume the input is smaller than that.

-- Also, i suppose if ndim == 2 and padding_size == 0, this would trigger? Is that the intent here?
In that case, input.ndim() * 2 equals padding_size + 4 which would not trigger the exception.  But it's wrong because the op. assumes 1 dimension for batch; 1 dimension for channels, and 1 or more channels for dimensions.  I should add such a check.